src/components/table/table.tsx

```diff
@@ -113,16 +128,29 @@ export function Table<T extends object>({
           [state, controlledState]
         )
       },
-      groupByFn,
+      toggleSelectedItemClb,
+      itemIsDisabled,
     },
     ...reactTableHooks
   )
 
   useEffect(() => {
-    if (selectedItemsClb) {
-      selectedItemsClb(selectedFlatRows.map((r: Row<T>) => r.original))
+    if ((selectedFlatRows.length === 0 || isAllRowsSelected) && selectedItemsClb) {
+      const isGrouped = groupBy.length > 0
+      selectedItemsClb(
+        selectedFlatRows.reduce((h: Item[], r: Row<T>) => {
+          if (isGrouped && r.isGrouped) return h
+          if (itemIsDisabled(r)) return h
+          return [...h, r.original]
+        }, [])
+      )
     }
-  }, [selectedFlatRows, selectedItemsClb])
+  }, [selectedFlatRows, isAllRowsSelected, selectedItemsClb, groupBy, itemIsDisabled])
+
+  useEffect(() => {
```

Comment:
I'm a bit confused with this effect. If we get `isAllRowsExpanded`, and it is falsy, we make them `expanded` by an imperative command, right? Why?